lib/fluent-plugin-s3-arrow/schemas/aws_glue.rb

```diff
@@ -0,0 +1,113 @@
+require 'aws-sdk-glue'
+
+module FluentPluginS3Arrow
+  module Schemas
+    class AWSGlue
+      class Error < RuntimeError; end
+      class ConvertError < Error; end
+      class Field < Struct.new(:name, :type); end
+
+      def initialize(table_name, options={})
+        @table_name = table_name
+        @database_name = options.delete(:database_name) || "default"
+        @catalog_id = options.delete(:catalog_id)
+        @client = Aws::Glue::Client.new(options)
+      end 
+
+      def to_arrow
+        glue_schema = fetch_glue_schema
+        convert_to_arrow_schema(glue_schema)
+      end
+
+      private
+      def fetch_glue_schema
+        glue_table = @client.get_table({
+          catalog_id: @catalog_id,
+          database_name: @database_name,
+          name: @table_name
+        })
+        glue_table.table.storage_descriptor.columns
+      end
+
+      def convert_to_arrow_schema(glue_schema)
+        glue_schema.map do |glue_field|
+          convert_to_arrow_field(glue_field)
+        end
```

Comment:
The method name says Apache Arrow schema but this doesn't return Apache Arrow schema:

```suggestion
        arrow_fields = glue_schema.map do |glue_field|
          convert_to_arrow_field(glue_field)
        end
        Arrow::Schema.new(arrow_fields)
```

Reply:
Ah, we say `{...}` not `Arrow::{Field,Schema}` as `arrow_` in this class.
How about saying `{...}` as `arrow_{field,schema}_description`? It's called `description` in Red Arrow: https://github.com/apache/arrow/blob/master/ruby/red-arrow/lib/arrow/field.rb#L32

Reply:
We now use the term `arrow_{field,schema}_description`. 101ebae